README.md

```diff
@@ -138,19 +138,58 @@ interface "wasi:kv/data/set-many" {
   set-many: func(key_values: stream<(key, stream<u8>)>) -> result<_, Error>
 }
 
+interface "wasi:kv/data/range" {
+  // request using a range of keys
+  use { Error, payload, key-range, key-value } from "wasi:kv/types"
+
+  // returns keys within the range
+  get-keys: func(range: key-range) -> result<stream<key>, Error>
+
+  // returns key-value tuples within the key range
+  get-range: func(range: key-range) -> Result<stream<key-value>, Error>
+
+  // returns the number of keys that match the range
+  count-range: func(range: key-range) -> result<u64, Error>
```

Comment:
While I get the motivation behind the performance optimization here, e.g. getting the count rather than marshaling the contents and then counting the items, this can be particularly problematic and can lead to edge cases where people assume it works one way and it doesn't, or worse, the behavior of a range count differs from one implementation to the next.

Reply:
It's more than a performance optimization - it's a useful query in itself. e.g., how many users are online? how many documents are in my collection?

If there are undefined edge cases, we should tighten up the spec so that the outcome is determistic across providers. Isn't utf-8 sorting well-defined? I feel pretty strongly that the benefits of range scan outweigh extremely rare edge cases, especially since there are ways to avoid known edge cases
- an app can limit the character set of keys to avoid known problems
- any provider implementation that can't efficiently scan keys in a sorted order can decide not to implement this interface.